core/src/test/scala/unit/kafka/network/ConnectionQuotasTest.scala

```diff
@@ -329,15 +613,46 @@ class ConnectionQuotasTest {
   }
 
   // this method must be called on a separate thread, because connectionQuotas.inc() may block
+  private def acceptConnectionsAndVerifyRate(connectionQuotas: ConnectionQuotas,
+                                             listenerDesc: ListenerDesc,
+                                             numConnections: Long,
+                                             timeIntervalMs: Long,
+                                             expectedRate: Int,
+                                             epsilon: Int) : Unit = {
+    val startTimeMs = System.currentTimeMillis
+    acceptConnections(connectionQuotas, listenerDesc.listenerName, listenerDesc.defaultIp, numConnections, timeIntervalMs)
+    val elapsedSeconds = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis - startTimeMs)
+    val actualRate = (numConnections.toDouble / elapsedSeconds).toInt
+    if (actualRate - epsilon > expectedRate || actualRate + epsilon < expectedRate)
+      throw new TestFailedException(
+        (e: StackDepthException) =>
+          Some(s"Expected rate $expectedRate, but got $actualRate ($numConnections connections / $elapsedSeconds sec)"),
+        None,
+        Position.here)
```

Comment:
Can't we use `assertTrue` here to simplify?

Reply:
Yes, I had a memory that asserting from another thread would not do the right thing, but I think that was a very old memory. Checked and it works well. Replaced with `assertTrue`.